### ISSUE TITLE
May I ask how to generate netlists.txt based on the circuit diagram?

### ISSUE THREAD

**Issue by CP-Panizza** [jatinchowdhury18/R-Solver#2]
May I ask how to generate netlists.txt based on the circuit diagram?

**Comment by jatinchowdhury18** [comment]
Sure thing! I have written a little [blog post](https://medium.com/@jatinchowdhury18/wave-digital-filter-circuit-models-with-r-type-adaptors-39ad0ad658ce) describing the process of creating the netlist file for the "bassman.txt" example file. For a more complete discussion, I'd recommend reading Kurt Werner's [PhD Dissertation](https://stacks.stanford.edu/file/druid:jy057cz8322/KurtJamesWernerDissertation-augmented.pdf), as well as some of his other papers from the past 5+ years, which describe the process of developing R-Type adaptors in more detail.

**Comment by r3adne** [comment]
ik this is a super old issue, but i wanted to mention that a new possibility for an easier and more general decomposition algorithm was published in 2024, (since this thread) by Beattie and Ochs in doi: 10.1002/cta.3781

**Comment by jatinchowdhury18** [comment]
@r3adne Oh nice, that's a good find! If I have some time I might try auto-generating netlists using their approach, which could then be fed directly into the R-Solver. That said, I might keep that a little bit separate from this repo/code since I'd like to keep the scope of this code relatively small :).

**Comment by r3adne** [comment]
@jatinchowdhury18 absolutely, this isn't the place, I'm just scouring while working through my implementation of automated, self-lumping circuit graph to wave-digital structure. Unfortunately this can't be open source.
